command/agent/config/config.go

```diff
@@ -229,10 +229,7 @@ func LoadConfig(path string) (*Config, error) {
 		}
 
 		if result.Cache.UseAutoAuthToken {
-			if result.AutoAuth == nil {
-				return nil, fmt.Errorf("cache.use_auto_auth_token is true but auto_auth not configured")
-			}
-			if result.AutoAuth.Method.WrapTTL > 0 {
+			if result.AutoAuth != nil && result.AutoAuth.Method.WrapTTL > 0 {
```

Comment:
```suggestion
			if result.AutoAuth != nil && result.AutoAuth.Method != nil && result.AutoAuth.Method.WrapTTL > 0 {
```

Reply:
This is a really good point. I don't think we have to worry about checking `AutoAuth.Method == nil`, since we're guaranteed by the parser to have a non-nil method if `AutoAuth` is non-nil.